lib/streamlit/runtime/secrets.py

```diff
@@ -45,33 +44,47 @@ def _missing_key_error_message(key: str) -> str:
     )
 
 
-class AttrDict(UserDict):  # type: ignore[type-arg]
+class AttrDict(Mapping[str, Any]):
     """
     We use AttrDict to wrap up dictionary values from secrets
     to provide dot access to nested secrets
     """
 
+    def __init__(self, value):
+        self.__dict__.update(value)
+
     @staticmethod
     def _maybe_wrap_in_attr_dict(value) -> Any:
         if not isinstance(value, dict):
```

Comment:
nit: Checking for `dict` instead of `Mapping` is fine here, or?

Reply:
Done!